### PR TITLE
Moving to Kotlin

### PR DESCRIPTION
**Pull request by yenerm**
Activities changed to fragments, Navigation implemented, java code is converted to kotlin

### CODE REVIEW

File: BasicAndroidAccessibility/app/build.gradle [hunk 1]
```diff
@@ -1,4 +1,6 @@
 apply plugin: 'com.android.application'
+apply plugin: 'kotlin-android'
+apply plugin: 'kotlin-android-extensions'
```
Comment (ceruleanotter) [inline BasicAndroidAccessibility/app/build.gradle]: If you're not using this plugin, remove

File: BasicAndroidAccessibility/app/src/main/java/com/example/android/basicandroidaccessibility/ContentGroupingFragment.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
+// Copyright 2016 Google Inc.
```
Comment (ceruleanotter) [inline BasicAndroidAccessibility/app/src/main/java/com/example/android/basicandroidaccessibility/ContentGroupingFragment.kt]: Consider updating copyright

File: BasicAndroidAccessibility/app/src/main/java/com/example/android/basicandroidaccessibility/HomeFragment.kt [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import androidx.navigation.fragment.findNavController
+
+
+class HomeFragment : Fragment() {
+
+
```
Comment (ceruleanotter) [inline BasicAndroidAccessibility/app/src/main/java/com/example/android/basicandroidaccessibility/HomeFragment.kt]: Remove extra space

File: BasicAndroidAccessibility/app/src/main/java/com/example/android/basicandroidaccessibility/HomeFragment.kt [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
+        // Inflate the layout for this fragment
+        return inflater.inflate(R.layout.fragment_home, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+
```
Comment (ceruleanotter) [inline BasicAndroidAccessibility/app/src/main/java/com/example/android/basicandroidaccessibility/HomeFragment.kt]: Remove extra space

File: BasicAndroidAccessibility/app/src/main/java/com/example/android/basicandroidaccessibility/ContentLabelingFragment.kt [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+        updateImageButton()
+        playPauseToggleImageView.setOnClickListener {
+            playing = !playing
+            updateImageButton()
+        }
+
+    }
+
+
```
Comment (ceruleanotter) [inline BasicAndroidAccessibility/app/src/main/java/com/example/android/basicandroidaccessibility/ContentLabelingFragment.kt]: Remove extra space

File: BasicAndroidAccessibility/app/src/main/java/com/example/android/basicandroidaccessibility/HomeFragment.kt [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        val buttonContentLabeling = view.findViewById<Button>(R.id.content_labeling_button)
+        buttonContentLabeling?.setOnClickListener {
+            findNavController().navigate(R.id.action_homeFragment_to_contentLabelingFragment, null)
```
Comment (ceruleanotter) [inline BasicAndroidAccessibility/app/src/main/java/com/example/android/basicandroidaccessibility/HomeFragment.kt]: Use convenience method of generating navigation click listeners:
https://developer.android.com/reference/kotlin/androidx/navigation/Navigation.html#createNavigateOnClickListener(kotlin.Int)

File: BasicAndroidAccessibility/app/src/main/java/com/example/android/basicandroidaccessibility/HomeFragment.kt [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+        val buttonExpandTouchArea = view.findViewById<Button>(R.id.expand_touch_area_button)
+        buttonExpandTouchArea?.setOnClickListener {
+            findNavController().navigate(R.id.action_homeFragment_to_expandTouchAreaFragment, null)
+        }
+        val buttonInsufficientContrast = view.findViewById<Button>(R.id.insufficient_contrast_button)
+        buttonInsufficientContrast?.setOnClickListener {
+            findNavController().navigate(R.id.action_homeFragment_to_insufficientContrastFragment, null)
+        }
+
+
```
Comment (ceruleanotter) [inline BasicAndroidAccessibility/app/src/main/java/com/example/android/basicandroidaccessibility/HomeFragment.kt]: remove extra space

File: BasicAndroidAccessibility/app/src/main/java/com/example/android/basicandroidaccessibility/LiveRegionFragment.kt [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+                radioButton.textSize = 18f
+                radioGroup.addView(radioButton)
+            }
+
+            radioGroup.setOnCheckedChangeListener { _, checkedId ->
+                val indexOfCheckedChild = radioGroup.indexOfChild(view.findViewById<View>(checkedId))
+                when (indexOfCheckedChild) {
+                    correctAnswerIndex -> {
+                        feedbackTextView.setText(R.string.correct)
+                        feedbackTextView.setBackgroundColor(ContextCompat.getColor(context!!,
```
Comment (ceruleanotter) [inline BasicAndroidAccessibility/app/src/main/java/com/example/android/basicandroidaccessibility/LiveRegionFragment.kt]: !! is a code smell, is there a better way to do this?

File: BasicAndroidAccessibility/app/src/main/java/com/example/android/basicandroidaccessibility/ExpandTouchAreaFragment.kt [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageButton
+import android.widget.RelativeLayout
+import android.widget.TextView
+import androidx.appcompat.widget.Toolbar
```
Comment (ceruleanotter) [inline BasicAndroidAccessibility/app/src/main/java/com/example/android/basicandroidaccessibility/ExpandTouchAreaFragment.kt]: Did you clean up imports? I don't immediately see a Toolbar

File: BasicAndroidAccessibility/app/src/main/java/com/example/android/basicandroidaccessibility/ExpandTouchAreaFragment.kt [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+
+
+class ExpandTouchAreaFragment : Fragment() {
+
+    private lateinit var toggleImageButton: ImageButton
+    private var playing:Boolean = false
```
Comment (ceruleanotter) [inline BasicAndroidAccessibility/app/src/main/java/com/example/android/basicandroidaccessibility/ExpandTouchAreaFragment.kt]: spacing

File: BasicAndroidAccessibility/app/src/main/res/layout/activity_main.xml [hunk 11]
```diff
@@ -13,51 +13,29 @@
...
+    android:fitsSystemWindows="true"
+    tools:context=".MainActivity">
 
-    <Button
-        android:id="@+id/activity_content_grouping_button"
-        android:layout_width="match_parent"
-        android:layout_height="wrap_content"
-        android:onClick="launchActivity"
-        android:text="@string/title_activity_content_grouping" />
+        <androidx.appcompat.widget.Toolbar
```
Comment (ceruleanotter) [inline BasicAndroidAccessibility/app/src/main/res/layout/activity_main.xml]: In your activity to set up the toolbar, use Navigation UI's methods to set it up:
https://developer.android.com/reference/kotlin/androidx/navigation/ui/NavigationUI#setupActionBarWithNavController(androidx.appcompat.app.AppCompatActivity,%20androidx.navigation.NavController)

File: BasicAndroidAccessibility/app/build.gradle (hunk 1 above)
Comment (yenerm) [reply BasicAndroidAccessibility/app/build.gradle]: in use, crashes when i remove them.

File: BasicAndroidAccessibility/app/src/main/java/com/example/android/basicandroidaccessibility/LiveRegionFragment.kt [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+        super.onViewCreated(view, savedInstanceState)
+
+        val androidVersions = ArrayList(Arrays.asList(*resources.getStringArray(
+                R.array.android_versions)))
+        val correctAnswerIndex = androidVersions.indexOf(getString(R.string.lollipop))
+        val radioGroup = view.findViewById<RadioGroup>(R.id.radio_group)
+        val feedbackTextView = view.findViewById<TextView>(R.id.feedback_text_view)
+
+        if (radioGroup != null && correctAnswerIndex != -1) {
+            for (i in androidVersions.indices) {
```
Comment (tiembo) [inline BasicAndroidAccessibility/app/src/main/java/com/example/android/basicandroidaccessibility/LiveRegionFragment.kt]: ```androidVersions.indices.forEach { i ->```

File: BasicAndroidAccessibility/app/src/main/java/com/example/android/basicandroidaccessibility/LiveRegionFragment.kt [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+                            feedbackTextView.setBackgroundColor(ContextCompat.getColor(it,
+                                    R.color.red))
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
```
Comment (tiembo) [inline BasicAndroidAccessibility/app/src/main/java/com/example/android/basicandroidaccessibility/LiveRegionFragment.kt]: formatting

File: BasicAndroidAccessibility/app/src/main/java/com/example/android/basicandroidaccessibility/MainActivity.kt [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.android.basicandroidaccessibility
+
+import android.content.res.Resources
```
Comment (tiembo) [inline BasicAndroidAccessibility/app/src/main/java/com/example/android/basicandroidaccessibility/MainActivity.kt]: remove unused imports

File: BasicAndroidAccessibility/app/src/main/java/com/example/android/basicandroidaccessibility/MainActivity.kt [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+import androidx.navigation.ui.setupWithNavController
+
+class MainActivity : AppCompatActivity() {
+    private lateinit var appBarConfiguration : AppBarConfiguration
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        val toolbar = findViewById<Toolbar>(R.id.toolbar)
```
Comment (tiembo) [inline BasicAndroidAccessibility/app/src/main/java/com/example/android/basicandroidaccessibility/MainActivity.kt]: Lines 35-45 can be simplified:
```
        val navController = Navigation.findNavController(this, R.id.my_nav_host_fragment)
        appBarConfiguration = AppBarConfiguration(navController.graph)

        val toolbar: Toolbar = findViewById(R.id.toolbar)
        setSupportActionBar(toolbar)
        toolbar.setupWithNavController(navController, appBarConfiguration)
```

File: BasicAndroidAccessibility/app/build.gradle [hunk 16]
```diff
@@ -21,4 +23,9 @@ android {
 dependencies {
     implementation 'androidx.appcompat:appcompat:1.0.2'
     implementation 'com.google.android.material:material:1.0.0'
+
+    //Navigation
+    implementation "androidx.navigation:navigation-fragment-ktx:$rootProject.navigationVersion"
+    implementation "androidx.navigation:navigation-ui-ktx:$rootProject.navigationVersion"
+    implementation 'androidx.legacy:legacy-support-v4:1.0.0'
```
Comment (tiembo) [inline BasicAndroidAccessibility/app/build.gradle]: unused, ok to remove

File: BasicAndroidAccessibility/app/build.gradle [hunk 17]
```diff
@@ -21,4 +23,9 @@ android {
 dependencies {
     implementation 'androidx.appcompat:appcompat:1.0.2'
     implementation 'com.google.android.material:material:1.0.0'
```
Comment (tiembo) [inline BasicAndroidAccessibility/app/build.gradle]: unused, ok to remove

**Pull request merged**